intents/oversights/simpsons_paradox.py

```diff
@@ -0,0 +1,292 @@
+    """
+    Args:
+        initial_result_table: Type-pandas.dataframe
+            It has the contents of the csv file
+        new_result_table: Type-pandas.dataframe
+            It has the contents of the csv file
+        new_added_column: Type-string
+            The new column added in the initial grouping column.
+        slice1: Type-string
+            the first slice by which we do comparision.
+        dimensions_len: Type-integer
+            size of the initial gfrouping dimensions.
+
+    Returns:
+        return a dictionary where each dictionary represent a debiasing 
+               suggestion according to the new column.
+    """
+    initial_num_rows = initial_result_table.shape[0]
+    new_num_rows = new_result_table.shape[0]
+
+    # changing dataframe to list of list to do next set of operation
+    initial_table_matrix = initial_result_table.values.tolist()
+    new_table_matrix = new_result_table.values.tolist()
+
+    initial_row_i = 0
+    new_row_i = 0
+
+    suggestion_row_list = []
+
+    """ dominant percentage is the percentage of pairs where 
+        (first value - second value) is positive."""
+
+    while initial_row_i < initial_num_rows:
+
+        # check if it is the last row or there is only one slice for this group.
+        if initial_row_i == initial_num_rows - 1 or initial_table_matrix[initial_row_i][:dimensions_len] != initial_table_matrix[initial_row_i + 1][:dimensions_len]:
+
+            initial_difference = 0
+            if initial_table_matrix[initial_row_i][dimensions_len] == slice1:
+                initial_difference = initial_table_matrix[initial_row_i][dimensions_len + 1]
+            else:
+                initial_difference = -initial_table_matrix[initial_row_i][dimensions_len + 1]
+            
+            # it stores the dominent percent for the initial table
+            initial_dominant_percent = 0
+            if initial_difference > 0:
+                initial_dominant_percent = 100
+
+            positive_count = 0
+            total_count = 0
+
+            # while the group data is same for both the table it will continue in the loop
+            while new_row_i < new_num_rows and initial_table_matrix[initial_row_i][:dimensions_len] == new_table_matrix[new_row_i][:dimensions_len] :
+                new_difference = 0
+                if new_table_matrix[new_row_i][dimensions_len + 1] == slice1:
+                    new_difference = new_table_matrix[new_row_i][dimensions_len + 2]
+                else:
+                    new_difference = -new_table_matrix[new_row_i][dimensions_len + 2]
+                
+                if new_difference > 0:
+                    positive_count = positive_count + 1
+                total_count = total_count + 1
+
+                new_row_i = new_row_i + 1
+
+            new_dominant_percent = (positive_count / total_count) * 100
+
+            # Finally according to the difference between dominent percentage we shall
```

Comment:
dominent -> dominant

Reply:
fixed this issue here and at 2 more places